avaje-config/src/main/java/io/avaje/config/InitialLoader.java

```diff
@@ -188,12 +189,14 @@ private boolean loadTest() {
   }
 
   /**
-   * Load configuration defined by a <em>load.properties</em> entry in properties file.
+   * Recursively Load configuration defined by a <em>load.properties</em> entry in properties file.
    */
   private void loadViaIndirection() {
     String paths = loadContext.indirectLocation();
-    if (paths != null) {
+    if (paths != null && !loadProperties.contains(paths)) {
       loadViaPaths(paths);
```

Comment:
See here is the BFS part.

You are loading all the top level property resources here.

What you need to do is pull each path out and recurse on that. 

Such that `last.properties` is really last.